lib/post_revisor.rb

```diff
@@ -405,6 +405,7 @@ def is_last_post?
   def plugin_callbacks
     DiscourseEvent.trigger(:before_edit_post, @post)
     DiscourseEvent.trigger(:validate_post, @post)
+    DiscourseEvent.trigger(:before_edit_topic, @topic, @editor) if topic_changed?
```

Comment:
I think `:before_edit_post` is sufficient here and we don't need another event trigger. We can easily check if it is a topic by checking if the `post_number` is one.